packages/grid/x-data-grid/src/tests/components.DataGrid.test.tsx

```diff
@@ -64,6 +55,29 @@ describe('<DataGrid /> - Components', () => {
       expect(getCell(0, 0)).to.have.attr('data-name', 'foobar');
     });
 
+    it('should not override cell dimensions when passing `slotProps.cell.style` to the cell', () => {
```

Comment:
@romgrk Could you cherry-pick this test to the sticky headers PR?

Reply:
Yeah but I don't understand what it's testing? Aren't the styles empty both times? Also we're changing a bit the approach in sticky, we use CSS variables to set dimensions.

Reply:
> Yeah but I don't understand what it's testing? Aren't the styles empty both times?

It doesn't matter what we pass in `slotProps.cell.style` (this is why it's an empty object).
What matters is that the cell shouldn't lose its dimensions which are set as inline styles (see the screenshot in https://github.com/mui/mui-x/pull/11215#issue-2012794407)

> we're changing a bit the approach in sticky, we use CSS variables to set dimensions.

The CSS variables are passed as inline styles, so this test still makes sense there. If `slotProps.cell.style` is not properly merged with the internal inline styles, they will be overwritten 🙂

Reply:
I don't like that test :| I think we should move the destructured props from this:

```tsx
<div
  style={style}
  {...other}
>
  {children}
</div>
```

to this and call it a day:

```tsx
<div
  {...other}
  style={style}
>
  {children}
</div>
```

Probably because I just spent weeks going through all sorts of tests, but I feel like our test suite might need some cleanup. And I don't like the idea that we keep adding tests for bugs that are very unlikely to reoccur, such as this one, because we keep paying more and more time to run it. And if we test for the `style` prop, why wouldn't we also test all the other props that are equally likely to be overriden?

Lmk what you think.

Reply:
On a different note, about the test, IIUC this might be incorrect. We want to test that we preserve the cell dimensions but we never have a moment when `slotProps.cell.style` is `undefined`, therefore `initialCellWidth` will always return the same value, regardless if the dimensions are correct.

Reply:
> And I don't like the idea that we keep adding tests for bugs that are very unlikely to reoccur, such as this one, because we keep paying more and more time to run it.

Unit tests are pretty fast, so I don't mind the "time to run" cost.
I believe the biggest value of regression tests like this one is when we do major refactors (like sticky headers), where it's hard to foresee all potential issues (if the tests do no not test implementation details).
Weren't they helpful?

Reply:
> We want to test that we preserve the cell dimensions but we never have a moment when slotProps.cell.style is undefined,

Hmm, the initial render doesn't have any cell styles passed, so the cell width is the proper one.
After the button click, the cell styles are added. Am I missing something?

Maybe `setProps` instead of button click would be more readable.

Reply:
I double checked and the test fails if the fix is not applied:

> AssertionError: expected 48.609375 to equal 100

Reply:
Fair point, let's keep it.

Reply:
Thanks!